substrate/pkg/controller/substrate/cluster/config.go

```diff
@@ -250,8 +250,11 @@ After=docker.service iptables-restore.service
 Requires=docker.service
 
 [Service]
-ExecStart=/usr/bin/kubelet --cluster-dns=10.96.0.10 --cluster-domain=cluster.local --hostname-override=%s --pod-manifest-path=/etc/kubernetes/manifests --kubeconfig=/etc/kubernetes/kubelet.conf  --cgroup-driver=systemd  --container-runtime=docker --network-plugin=cni --pod-infra-container-image=public.ecr.aws/eks-distro/kubernetes/pause:v1.18.9-eks-1-18-1  --provider-id=aws:///%s/%s --node-labels=kit.aws/substrate=control-plane
-Restart=always`, substrate.Name, aws.StringValue(instancesOutput.Reservations[0].Instances[0].Placement.AvailabilityZone), aws.StringValue(substrate.Status.Infrastructure.MasterInstanceID))), 0644); err != nil {
+ExecStart=/usr/bin/kubelet --cluster-dns=10.96.0.10 --cluster-domain=cluster.local --hostname-override=%s --pod-manifest-path=/etc/kubernetes/manifests --kubeconfig=/etc/kubernetes/kubelet.conf  --cgroup-driver=systemd  --container-runtime=docker --network-plugin=cni --pod-infra-container-image=public.ecr.aws/eks-distro/kubernetes/pause:v1.24.10-eks-1-24-12  --provider-id=aws:///%s/%s --node-labels=kit.aws/substrate=control-plane
+Restart=always
+
+[Install]
+WantedBy=multi-user.target`, substrate.Name, aws.StringValue(instancesOutput.Reservations[0].Instances[0].Placement.AvailabilityZone), aws.StringValue(substrate.Status.Infrastructure.MasterInstanceID))), 0644); err != nil {
```

Comment:
IIUC this would ensure it will help systemd to re-start it when instance reboots, which I think is the bug Prateek is majorly trying to fix through this PR ?